### PR TITLE
Deprecate `Field.include`

### PR DESCRIPTION
**Pull request by hramezani**
Selected Reviewer: @Kludex

### CODE REVIEW

**Comment by hramezani** [comment]
please review

File: docs/usage/serialization.md [hunk 1]
```diff
@@ -705,12 +705,12 @@ print(t.model_dump(exclude={'value': True, 'user': {'username'}}))
...
 ```py
-from pydantic import BaseModel, Field, SecretStr
+from pydantic import BaseModel, SecretStr
 
 
 class User(BaseModel):
-    id: int = Field(..., include=True)
-    username: str = Field(..., include=True)  # overridden by explicit include
+    id: int
+    username: str  # overridden by explicit include
```
Comment (Kludex) [inline docs/usage/serialization.md:713]: Does this makes sense still?

File: pydantic/fields.py [hunk 2]
```diff
@@ -793,6 +785,10 @@ def Field(  # noqa: C901
     if validation_alias in (_Unset, None):
         validation_alias = alias
 
+    include = extra.pop('include', None)  # type: ignore
+    if include is not None:
+        warn('`include` is deprecated and does nothing. It will be removed, use `exclude` instead', DeprecationWarning)
```
Comment (Kludex) [inline pydantic/fields.py:790]: AFAIU, this is not a deprecation... If the functionality is removed, it's a removal. The parameter itself is deprecated.

Also, how is `exclude` a replacement for `include`? 🤔

Comment (hramezani) [reply pydantic/fields.py:790]: > AFAIU, this is not a deprecation... If the functionality is removed, it's a removal. The parameter itself is deprecated.
> 

I think it is a deprecation as it did nothing before and still does nothing. So, no change in functionality. just warning the user.
 
> Also, how is `exclude` a replacement for `include`? 🤔

`include=False` can be replaced with `exclude=True`. That's why I suggested using `exclude`.

File: docs/usage/serialization.md (hunk 1 above)
Comment (hramezani) [reply docs/usage/serialization.md]: No, I've just removed.

**Review by Kludex — APPROVED** [review]

**Pull request merged**